server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -127,6 +137,12 @@ public ClientQuerySegmentWalker(
     this.serverConfig = serverConfig;
     this.cache = cache;
     this.cacheConfig = cacheConfig;
+    this.subqueryLimitUtils = new SubqueryLimitUtils(
```

Comment:
I made the `SubqueryLimitUtils` class non-static since we compute the size of the lookup at the start, and therefore it's required to have an internal state. Perhaps renaming it to `SubqueryLimitHelper` would be more appropriate.